### PR TITLE
fix(Core/Spells): Implement SPELL_EFFECT_ACTIVATE_OBJECT

### PR DESCRIPTION
**Pull request by Nefertumm**
Co-authored-by: jackpoz <giacomopoz@gmail.com>

## Changes Proposed:
-  Properly implement the spell effect and the dbc usage.
-  https://github.com/TrinityCore/TrinityCore/pull/24997

## Issues Addressed:

- Closes none

## SOURCE:

Trinity.

## Tests Performed:

- Tested in-game with spell 24082-24083 in Zul Gurub.

## How to Test the Changes:

1. Lookup for spells with the effect SPELL_EFFECT_ACTIVATE_OBJECT and test them.
2. If there are any script which replaces the effect, remove it and test the spell again and see if it works.

## Known Issues and TODO List:

- [ ] Our gameobjects are utterly broken. When I tested with the spell ids above, the gameobject didn't despawn whatsoever (and also, the spiders were spawned on player when they are supposed to spawn on gob pos).
- [x] This is needed for https://github.com/azerothcore/azerothcore-wotlk/pull/11647

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1651816895816737900.sql [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  `Texture_3` INT NOT NULL DEFAULT 0,
+  `Attach_Model_1` INT NOT NULL DEFAULT 0,
+  `Attach_Model_2` INT NOT NULL DEFAULT 0,
+  `Attach_Model_3` INT NOT NULL DEFAULT 0,
+  `Attach_Model_4` INT NOT NULL DEFAULT 0,
+  PRIMARY KEY (`ID`)
+) ENGINE=MYISAM DEFAULT CHARSET=utf8mb4;
+
+-- Note: All of these should be targetable by spells 46904 and 46903, but conditions are only set for Stormwind (damn Horde fanatics)
+UPDATE gameobject_template_addon SET artkit0 = 121, artkit1 = 122 WHERE entry IN (
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1651816895816737900.sql:24]: ticks

File: src/server/game/Spells/SpellEffects.cpp [hunk 2]
```diff
@@ -4266,25 +4266,82 @@ void Spell::EffectSummonPlayer(SpellEffIndex /*effIndex*/)
...
+        case GameObjectActions::UseArtKit3:
+        {
+            GameObjectTemplateAddon const* templateAddon = gameObjTarget->GetTemplateAddon();
+
+            uint32 artKitIndex = uint32(action) - uint32(GameObjectActions::UseArtKit0);
 
-    //ScriptInfo activateCommand;
-    //activateCommand.command = SCRIPT_COMMAND_ACTIVATE_OBJECT;
+            uint32 artKitValue = 0;
+            if (templateAddon != nullptr)
```
Comment (Kitzunu) [inline src/server/game/Spells/SpellEffects.cpp:4328]: remove" != nullptr"

**Comment by Nyeriah** [comment]
Needs to be force merged

**Comment by Nyeriah** [comment]
Needs to be force merged

**Review by acidmanifesto — APPROVED** [review]

**Pull request merged**